### PR TITLE
net: lib: lwm2m: add new sensor objects

### PR DESCRIPTION
**Pull request by rerickson1**
Add generic, humidity and pressure sensor objects.

### CODE REVIEW

**Review by jukkar — COMMENTED** [review]
Please remove the `subsys: ` from commit subject, it is not needed.

File: subsys/net/lib/lwm2m/Kconfig.ipso [hunk 1]
```diff
@@ -39,6 +39,82 @@ config LWM2M_IPSO_TEMP_SENSOR_TIMESTAMP
...
+	  Sensor instances available to the LWM2M client.
+
+config LWM2M_IPSO_GENERIC_SENSOR_TIMESTAMP
+	bool "Add timestamp resources to IPSO objects"
+	default y
+	depends on LWM2M_IPSO_GENERIC_SENSOR && LWM2M_IPSO_TIMESTAMP_EXTENSIONS
+	help
+	  Add a non-standard timestamp resource to each object.
+
+config LWM2M_IPSO_GENERIC_SENSOR_IPSO_OBJECT_ID
```
Comment (rlubos) [inline subsys/net/lib/lwm2m/Kconfig.ipso:62]: I'm a bit confused about this config, isn't Generic Sensor IPSO object supposed to have a fixed ID (3300) as it's defined in the registry?
https://raw.githubusercontent.com/OpenMobileAlliance/lwm2m-registry/prod/3300.xml

File: subsys/net/lib/lwm2m/ipso_pressure_sensor.c [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+static struct lwm2m_engine_obj sensor;
+static struct lwm2m_engine_obj_field fields[] = {
+	OBJ_FIELD_DATA(SENSOR_VALUE_RID, R, FLOAT32),
+	OBJ_FIELD_DATA(UNITS_RID, R_OPT, STRING),
+	OBJ_FIELD_DATA(MIN_MEASURED_VALUE_RID, R_OPT, FLOAT32),
+	OBJ_FIELD_DATA(MAX_MEASURED_VALUE_RID, R_OPT, FLOAT32),
+	OBJ_FIELD_DATA(MIN_RANGE_VALUE_RID, R_OPT, FLOAT32),
+	OBJ_FIELD_DATA(MAX_RANGE_VALUE_RID, R_OPT, FLOAT32),
+	OBJ_FIELD_EXECUTE_OPT(RESET_MIN_MAX_MEASURED_VALUES_RID),
```
Comment (rlubos) [inline subsys/net/lib/lwm2m/ipso_pressure_sensor.c:66]: What about optional "Current Calibration" (5821) and "Application Type" (5750) resources? They're specified in
https://raw.githubusercontent.com/OpenMobileAlliance/lwm2m-registry/prod/3323.xml

**Review by rlubos — COMMENTED** [review]
I think we should enable these new objects in the prj.conf file for compilation, just as others are:
https://github.com/zephyrproject-rtos/zephyr/blob/master/samples/net/lwm2m_client/prj.conf#L33
Otherwise, we won't even know if they compile after time as the CI won't build them.

File: subsys/net/lib/lwm2m/ipso_pressure_sensor.c (hunk 2 above)
Comment (rerickson1) [reply subsys/net/lib/lwm2m/ipso_pressure_sensor.c:61]: Following the implementation of temp_sensor, so not implementing these in the pressure sensor at this time.

**Comment by rerickson1** [comment]
changing back to `-INT32_MAX`. `ipso_temp_sensor.c` uses that as well.  Also, when testing with Leshan, `-INT32_MAX` is needed, `INT32_MIN` does not work properly.

File: include/net/lwm2m.h [hunk 3]
```diff
@@ -50,9 +50,12 @@
  * http://www.openmobilealliance.org/wp/OMNA/LwM2M/LwM2MRegistry.html
  */
 
+#define IPSO_OBJECT_GENERIC_SENSOR_ID		3300
```
Comment (rlubos) [inline include/net/lwm2m.h:53]: nit: This line has broken alignment (one tab missing).

File: include/net/lwm2m.h [hunk 4]
```diff
@@ -50,9 +50,12 @@
...
 #define IPSO_OBJECT_TEMP_SENSOR_ID			3303
+#define IPSO_OBJECT_HUMIDITY_SENSOR_ID      3304
```
Comment (rlubos) [inline include/net/lwm2m.h:55]: This one as well.

**Review by rlubos — COMMENTED** [review]
Thanks, LGTM, just a minor issue left.

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**